akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -386,6 +393,15 @@ object ByteString {
       }
     }
 
+    override def copyToArray[B >: Byte](dest: Array[B], start: Int, len: Int): Int = {
+      // min of the bytes available to copy, bytes there is room for in dest and the requested number of bytes
+      val copied = math.max(math.min(math.min(len, length), dest.length - start), 0)
```

Comment:
```suggestion
      val copied = math.min(math.min(len, length), dest.length - start)
```

Reply:
That `max` ensures that we never get a negative, even if start is past end, you saying that will not happen?

Reply:
So the behavior is that when start is past end, we do not pass anything and return '0'.

We can also achieve that by not doing the 'max' here but returning '0' when `!(copied > 0)`

Reply:
Yep, so we'd avoid one static method call (probably inlined anyways)?